modules/xmpp/ChatRoom.js

```diff
@@ -885,12 +902,15 @@ export default class ChatRoom extends Listenable {
                     .length) {
             this.discoRoomInfo();
         }
+        const jsonMessage = $(msg).find('>json-message').text();
+        const parsedJson = tryParseJSONAndVerify(jsonMessage);
 
-        const jsonPayload = tryParseJSONAndVerify(txt);
-
-        if (jsonPayload) {
+        // We emit this event if the message is a valid json, and is not
+        // delivered after a delay, i.e. stamp is undefined.
+        // e.g. - subtitles should not be displayed if delayed.
+        if (parsedJson && stamp === undefined) {
```

Comment:
@bgrozev , can you confirm if any other reasons are there for checking if stamp is undefined ?